include/libtorrent/extensions.hpp

```diff
@@ -192,6 +192,11 @@ namespace libtorrent
 	struct peer_connection_handle;
 	struct torrent_handle;
 
+	typedef boost::function<bool(udp::endpoint const& source
+		, bdecode_node const& request, entry& response)> dht_extension_handler_t;
+
+	typedef std::vector<std::pair<std::string, dht_extension_handler_t> > dht_extensions_t;
```

Comment:
does this map an incoming DHT request to an extension handler? If so, should it be a map or boost::unordered_map?